docs/dev/ui_description.rst

```diff
@@ -0,0 +1,264 @@
+Glupyter Framework Overview
+===========================
+
+The glue-jupyter ("glupyter") package supports interacting with and
+visualizing data within the Jupyter environment using core elements from
+the glue python package. It is distinct because unlike the more
+prominent glue package, glupyter does not leverage Qt as the front-end
+GUI library. Instead, glupyter maintains the separation between the data
+model (e.g. the core elements from glue that are not dependent on the
+front-end library), and the view of the data (in this case, web-based
+tooling provided by Jupyter).
+
+Glupyter implements a base ``JupyterApplication`` object through which
+users can manage their data, create viewers, and add links between data
+sets. The data management and linking are controlled separate from the
+viewers in that changes made directly to the data state propagate to the
+UI -- that is, the UI does not contain any state, which allows the application to be easily controlled from code. The viewers themselves are based on the
+IPyWidget package which allows the creation of widgets that can be used
+and interacted with in python, but rendered in a browser environment.
+
+There are two distinct use cases for the glupyter environment:
+
+    1. as a means to procedurally interact with pieces of a user's workflow
+    *in addition* to their work in e.g. a Jupyter Notebook;
+    2. to provide users a web-based GUI to interact with and visualize their
+    data while hiding the Python code, e.g. a standalone web application.
+
+These two use cases describe a Python-first and GUI-first approach,
+respectively. This document will focus on detailing the design of the
+GUI-first approach, depicted in the following diagram.
+
+.. note::
+    The ``.xml`` file in the ``img`` directory can be used to edit the
+    diagram using applications like, e.g.,
+    `draw.io <https://draw.io>`_.
+
+.. image:: ../img/glue-jupyter_diagram.png
+
+General user interface design
```

Comment:
I'll make it clearer in the text, but at the highest level, "jdaviz" is the collection of user plugins and specific viewers registered to glue-jupyter application, along with the declaration configuration. This isn't completely the case currently, because right now the `Application` class lives in jdaviz. Eventually, we'd want the `Application` case to live in glue-jupyter, and specific configurations would be fed to it to produce an "Application".

Functionally, "jdaviz"(/cubeviz/specviz/mosviz) would be the final "Application" product.